src/experiment.py

```diff
@@ -85,224 +119,248 @@ def drop_first_observations(self):
-        plt.hist2d(x, y, bins=numbins, range=[[-180,180],[-180,180]], cmap=plt.cm.Reds)
-        plt.colorbar(format=mtick.FormatStrFormatter("%.1e"))
-        plt.xlabel(self.columns[xcol])
-        plt.ylabel(self.columns[ycol])
-        plt.title(mytitle)
-        if plotname:
-            plot_filepath = f"{plotname}hist2D_{self.chain.replace(' ','')}_{xcol}_{ycol}_b{numbins}.png"
-            plt.savefig(plot_filepath, dpi=self.plot_dpi)
-            plt.clf()
-        else:
-            plt.show()
+        plot_filepath = os.path.join(plotdir,f"{self}_c{ycol}.png")
+        plt.savefig(plot_filepath, dpi=self.plot_dpi)
+        plt.clf()
         plt.close()
 
-    def plot_angle_histogram(self, angle_x, angle_y, plotname: str = None, numbins = 100):
+    def plot_angle_histogram(self, angle_x, angle_y, plotdir: str = "plots", numbins = 100):
         """
         Plots histogram fo angles for all subsequent mers in experiment (realisation)
         """
         x_data = np.array([])
         y_data = np.array([])
+        if angle_x.endswith('₁₃'):
+            x_columns_of_interest = [ f"{angle_x} mers {mer+1}, {mer+2}" for mer in range(self.no_mers-1) ]
+        else:
+            x_columns_of_interest = [ f"{angle_x} mer {mer+1}" for mer in range(self.no_mers) ]
 
-        x_columns_of_interest = [ f"{angle_x} mers {mer+1}, {mer+2}" for mer in range(self.no_mers) ]
-        y_columns_of_interest = [ f"{angle_y} mers {mer+1}, {mer+2}" for mer in range(self.no_mers) ]
+        if angle_y.endswith('₁₃'):
+            y_columns_of_interest = [ f"{angle_y} mers {mer+1}, {mer+2}" for mer in range(self.no_mers-1) ]
+        else:
+            y_columns_of_interest = [ f"{angle_y} mer {mer+1}" for mer in range(self.no_mers) ]
 
         x_cols = [ self.get_colnum_by_meaning(x) for x in x_columns_of_interest ]
         y_cols = [ self.get_colnum_by_meaning(x) for x in y_columns_of_interest ]
 
         for c in x_cols:
             x = self.dataframe.iloc[:, c]
-            x = correct_signs(x)
             x_data = np.concatenate((x_data, x))
 
         for c in y_cols:
             y = self.dataframe.iloc[:, c]
-            y = correct_signs(y)
             y_data = np.concatenate((y_data, y))
 
+        n_datapoints = min( len(x_data), len(y_data) )
+
         plt.subplots()
         plt.hist2d(x, y, bins=numbins, range=[[-180,180],[-180,180]], cmap=plt.cm.Reds)
-        plt.colorbar(format=mtick.FormatStrFormatter("%.1e"))
+        plt.colorbar(format=mtick.ScalarFormatter())
         plt.xlabel(angle_x)
         plt.ylabel(angle_y)
-        plt.title(f"Subsequent mers angles, R{self.num_realisation}")
-        if plotname:
-            plot_filepath = f"{plotname}hist2D_{self.chain.replace(' ','')}_{angle_x}_{angle_y}_b{numbins}.png"
-            plt.savefig(plot_filepath, dpi=self.plot_dpi)
-            plt.clf()
-        else:
-            plt.show()
+        plt.title(f"{self} - Subsequent mers angles, n={n_datapoints}")
```

Comment:
we can filter out analysis in f"{self}. We an also remove: "Subsequent mers angles" (I know I suggested it at the first place). The best  will be something like:   Albumin+CS6_realization:1_ion:Mg  or similar